src/gc.c

```diff
@@ -166,6 +166,60 @@ static int support_conservative_marking = 0;
 jl_gc_num_t gc_num = {0,0,0,0,0,0,0,0,0,0,0,0,0,0};
 static size_t last_long_collect_interval;
 
+// Helper functions to keep track of allocation and free amounts
+void jl_gc_count_allocd(void * addr, size_t sz, uint16_t tag) JL_NOTSAFEPOINT
+{
+#ifdef JULIA_ENABLE_THREADING
+    jl_atomic_fetch_add(&gc_num.allocd, sz);
```

Comment:
Please don't. This kills performance. The TODO in the code was wrong and this should be made thread local instead. The function that want to read these values is responsible for adding everything up. The only thread synchronization needed for the allocation thread should be to use relaxed atomic load and store and a normal increment.